### PR TITLE
mgr/orch: Properly handle NotImplementedError

### PR DESCRIPTION
**Pull request by sebastian-philipp**
They are masked as RuntimeError by the mgr.

Fixes: https://tracker.ceph.com/issues/44569

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/orchestrator/_interface.py [hunk 1]
```diff
@@ -1900,7 +1899,12 @@ def _oremote(self, meth, args, kwargs):
             raise NoOrchestrator()
 
         mgr.log.debug("_oremote {} -> {}.{}(*{}, **{})".format(mgr.module_name, o, meth, args, kwargs))
-        return mgr.remote(o, meth, *args, **kwargs)
+        try:
+            return mgr.remote(o, meth, *args, **kwargs)
+        except RuntimeError as e:
+            if meth != 'get_feature_set' and meth not in self.get_feature_set():
```
Comment (liewegas) [inline src/pybind/mgr/orchestrator/_interface.py]: why the special case for get_feature_set?  (maybe in a comment?)

**Comment by liewegas** [comment]
let's target this one at octopus?

File: src/pybind/mgr/orchestrator/_interface.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator/_interface.py]: `self.get_feature_set` calls `self._oremote`, which calls `self.get_feature_set` here.

**Comment by sebastian-philipp** [comment]
not yet needs-qa.

**Comment by liewegas** [comment]
retest this please

**Pull request merged**